src/StdLib.jl

```diff
@@ -117,4 +120,24 @@ Base.size(v::StdValArray) = (Int(cppsize(v)),)
 Base.getindex(v::StdValArray, i::Int) = cxxgetindex(v,i)[]
 Base.setindex!(v::StdValArray{T}, val, i::Int) where {T} = cxxsetindex!(v, convert(T,val), i)
 
+function StdDeque(v::Vector{T}) where {T}
+  return StdDeque{T}(v, length(v))
+end
+
+Base.IndexStyle(::Type{<:StdDeque}) = IndexLinear()
+Base.size(v::StdDeque) = (Int(cppsize(v)),)
+Base.getindex(v::StdDeque, i::Int) = cxxgetindex(v,i)[]
+Base.setindex!(v::StdDeque{T}, val, i::Int) where {T} = cxxsetindex(v, convert(T,val), i)
+Base.push!(v::StdDeque, x) = push_back(v, x)
+Base.pushfirst!(v::StdDeque, x) = push_front(v, x)
+Base.pop!(v::StdDeque) = pop_back(v)
+Base.popfirst!(v::StdDeque) = pop_front(v)
+Base.resize!(v::StdDeque, n::Integer) = resize(v, n)
+end
+
+# Dqueue iterator
```

Comment:
These are outside the module (after the `end` above), hence the undefined type error. The iterator interface is incorrect, you need to implement `Base.iterate`, see https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration